pkg/operator/k8sutil/resources.go

```diff
@@ -85,11 +86,32 @@ func (info *OwnerInfo) SetOwnerReference(object metav1.Object) error {
 	if err != nil {
 		return err
 	}
-	ownerRefs := append(object.GetOwnerReferences(), *info.ownerRef)
+	ownerRefs := object.GetOwnerReferences()
+	for _, v := range ownerRefs {
+		if referSameObject(v, *info.ownerRef) {
+			return nil
+		}
+	}
+	ownerRefs = append(ownerRefs, *info.ownerRef)
 	object.SetOwnerReferences(ownerRefs)
 	return nil
 }
 
+//This logic is copied from https://github.com/kubernetes-sigs/controller-runtime/blob/a905949b9040084f0c6d2a27ec70e77c3c5c0931/pkg/controller/controllerutil/controllerutil.go#L160
+func referSameObject(a, b metav1.OwnerReference) bool {
+	aGV, err := schema.ParseGroupVersion(a.APIVersion)
```

Comment:
More descriptive var names are preferred
```suggestion
groupVersionA, err := schema.ParseGroupVersion(a.APIVersion)
```